src/dune/lib.ml

```diff
@@ -1775,35 +1800,28 @@ module DB = struct
               |> Lib_info.of_local
             in
             match conf.public with
-            | None ->
-              [ (Dune_file.Library.best_name conf, Resolve_result.Found info) ]
+            | None -> [ (Dune_file.Library.best_name conf, Either.Left info) ]
             | Some p ->
               let name = Dune_file.Public_lib.name p in
               if Lib_name.equal name (Lib_name.of_local conf.name) then
-                [ (name, Found info) ]
+                [ (name, Either.Left info) ]
```

Comment:
FWIW, I think the names `Found`, `Redirect` here are better.

Reply:
Also, it looks like there's some type errors around here.

Reply:
They are better but they introduce quite a bit of a `assert false`. How about polymorphic variants perhaps? See the latest commit for how that would look like.

Reply:
Yeah, I like that better. 

Although maybe we should still respect the rule of no polymorphic variants, at least when no "killer features" of polymorphic variants are involved.
Why not:

```
type found_or_redirect =
  | Found of Lib_info.external_
  | Redirect of db option * (Loc.t * Lib_name.t)
```

Reply:
I got rid of the polymorphic variants.

Reply:
Thanks. Looks good!